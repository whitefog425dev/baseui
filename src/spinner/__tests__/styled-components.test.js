/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledSvg} from '../index.js';

describe('Spinner styled components', () => {
  test('StyledSvg - basic render', () => {
    const component = shallow(<StyledSvg />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledSvg has correct default styles',
    );
  });
  test('StyledSvg - custom size', () => {
    const component = shallow(<StyledSvg size={56} />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledSvg has correct height and width styles',
    );
  });
  test('StyledSvg - custom color', () => {
    const component = shallow(<StyledSvg color="#00FF00" />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledSvg has correct fill value',
    );
  });
});
