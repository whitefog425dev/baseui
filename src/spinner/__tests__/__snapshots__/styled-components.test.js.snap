// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner styled components StyledSvg - basic render: StyledSvg has correct default styles 1`] = `
Object {
  "animationDuration": "$theme.animation.timing700",
  "animationIterationCount": "infinite",
  "animationName": Object {
    "from": Object {
      "transform": "rotate(0deg)",
    },
    "to": Object {
      "transform": "rotate(360deg)",
    },
  },
  "animationTimingFunction": "linear",
  "cursor": "wait",
  "display": "inline-block",
  "fill": "$theme.colors.primary400",
  "height": "$theme.sizing.scale800",
  "width": "$theme.sizing.scale800",
}
`;

exports[`Spinner styled components StyledSvg - custom color: StyledSvg has correct fill value 1`] = `
Object {
  "animationDuration": "$theme.animation.timing700",
  "animationIterationCount": "infinite",
  "animationName": Object {
    "from": Object {
      "transform": "rotate(0deg)",
    },
    "to": Object {
      "transform": "rotate(360deg)",
    },
  },
  "animationTimingFunction": "linear",
  "cursor": "wait",
  "display": "inline-block",
  "fill": "$theme.colors.primary400",
  "height": "$theme.sizing.scale800",
  "width": "$theme.sizing.scale800",
}
`;

exports[`Spinner styled components StyledSvg - custom size: StyledSvg has correct height and width styles 1`] = `
Object {
  "animationDuration": "$theme.animation.timing700",
  "animationIterationCount": "infinite",
  "animationName": Object {
    "from": Object {
      "transform": "rotate(0deg)",
    },
    "to": Object {
      "transform": "rotate(360deg)",
    },
  },
  "animationTimingFunction": "linear",
  "cursor": "wait",
  "display": "inline-block",
  "fill": "$theme.colors.primary400",
  "height": "$theme.sizing.scale800",
  "width": "$theme.sizing.scale800",
}
`;
